pkg/controller/endpointslice/trigger_time_tracker.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslice
+
+import (
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+)
+
+// TriggerTimeTracker is a util used to compute the EndpointSliceLastChangeTriggerTime annotation which
+// is exported in the endpointSliceController's sync function.
+// See the documentation of the EndpointSliceLastChangeTriggerTime annotation for more details.
+//
+// Please note that this util may compute a wrong EndpointSliceLastChangeTriggerTime if the same object
+// changes multiple times between two consecutive syncs. We're aware of this limitation but we
+// decided to accept it, as fixing it would require a major rewrite of the endpointSlice controller and
+// Informer framework. Such situations, i.e. frequent updates of the same object in a single sync
+// period, should be relatively rare and therefore this util should provide a good approximation of
+// the EndpointSliceLastChangeTriggerTime.
+type TriggerTimeTracker struct {
```

Comment:
is it possible to refactor TriggerTimeTracker from `pkg/controller/endpoint/` to `pkg/controller/util`